### PR TITLE
Fix issue where no keys and values are sent to logr when using structured logging with json

### LINKED ISSUE #185: No keys and values are sent to logr when using structured logging with json
/kind bug

**What steps did you take and what happened:**
Tried using structured logging with json. None of the provided keys and values where logged.

**What did you expect to happen:**
All provided keys and values should be logged in json output.

**Anything else you would like to add:**
Pull-request coming

### PR DESCRIPTION
**Pull request by torresdal**
**What this PR does / why we need it**:
Fix issue where no keys and values are sent to logr when using structured logging with json.

**Which issue(s) this PR fixes**:
Fixes #185 

**Release note**:
```release-note
Fixed issue where no keys and values where passed on to logr when using structured logging with json
```

### CODE REVIEW

**Comment by torresdal** [comment]
Linked github account to cla

File: klog.go [hunk 1]
```diff
@@ -768,7 +768,7 @@ func (l *loggingT) errorS(err error, loggr logr.Logger, msg string, keysAndValue
 // if loggr is specified, will call loggr.Info, otherwise output with logging module.
 func (l *loggingT) infoS(loggr logr.Logger, msg string, keysAndValues ...interface{}) {
 	if loggr != nil {
-		loggr.Info(msg, keysAndValues)
+		loggr.Info(msg, keysAndValues...)
```
Comment (dims) [inline klog.go:771]: isn't this a problem in line 762 as well? can you please scan if there are other spots?

Comment (torresdal) [reply klog.go:771]: you're right @dims - missed that one. Did a scan, but couldn't find any more. Pull req updated with new commit.

**Comment by serathius** [comment]
LGTM, could you add tests?

**Comment by dims** [comment]
@torresdal looks like https://github.com/kubernetes/klog/pull/179 already filed and has a test case :) let's go with that.

/close

**Pull request closed without merging**